### ISSUE TITLE
The 'successor' concept for TS repos

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#606]
Could you guys spec it out a little bit more? What do you expect of it and how it should behave. @bgruening @nsoranzo @erasche

xref: https://github.com/galaxyproject/tools-iuc/pull/249

**Comment by hexylena** [comment]
Hmm. I'd see it as a "final" push to a repository, where you'd push a single file that looks like:

``` xml
<deprecation>
  <successor name="bowtie2" owner="iuc"  changeset="auto-filled-in" toolshed="auto-filled-in"/>
  <reason>
This repository was deprecated due to the much better wrapper of bowtie produced by the IUC.
  </reason>
</deprecation>
```

something simple like that?

**Comment by bgruening** [comment]
@erasche like it. Not sure if this needs to be pushable, but these information needs to be entered into the TS, somehow.
Tool updates should read this information and follow along to get the new version under different user name.

The most complicated part is probably to get the Tool linage correct, because it needs now to track different owners, repository names.

**Comment by nsoranzo** [comment]
I agree with @bgruening: it does not need to be a final commit and Galaxy should automatically let the user move to the other-user-repository as in normal repo upgrades.

**Comment by hexylena** [comment]
(my suggestion of a final commit is based around my hopeful view that 100%
of repository maintenance will be automated and part of git history, like
we did with repository metadata in the additional .shed.yml keys. Please
don't make me login to the TS...

tir. 18. aug. 2015, 19.31 skrev Nicola Soranzo notifications@github.com:

**Comment by nsoranzo** [comment]
@erasche: I was indeed thinking that we can added 2 new fields to `.shed.yml`:
- deprecated: boolean
- superseded-by: repository guid

These will be tracked in the Tool Shed database (and also in our git repository for tools-iuc and tools-devteam), but not in the Mercurial Tool Shed repository.

**Comment by hexylena** [comment]
@nsoranzo ah! Okay, that's fine then. Yes, agreed, it doesn't need to be tracked by mercurial. Just as long as we're tracking it in git/planemo is handling it. My apologies for the miscommunication here.
